git_history_of_project/git_history_of_project.md

```diff
@@ -0,0 +1,620 @@
+
+For example, to look back just one commit before the most recent checkpoint, use `git show HEAD~1`. If instead you wanted to look back three checkpoints (two commits before the most recent commit) you would enter `git show HEAD~2`.
+
+![The stack of 3 white flat boxes. Now the top box is labeled `HEAD`, the middle box `HEAD~1` and the bottom `HEAD~2`](media/Commit_stack_HEAD.svg)
+
+Using `HEAD` to refer to your commits can be great for looking at recent versions of your repository. But be careful, each time you commit to the repository, you are creating a new most-recent-version that is now the `HEAD` and every older version moves one layer down in the stack.
+
+![On the left, the stack of 3 flat white boxes are labeled `HEAD`, `HEAD~1`,  and `HEAD~2`. An arrow points to a stack on the right made of the 3 white boxes with a 4th red box now on top. This red box is labeled `HEAD` and the boxes below it are labeled, in order from top to bottom, `HEAD~1`, `HEAD~2`, `HEAD~3`](media/Commit_stack_HEAD2.svg)
+
+### Using the commit number
+
+If there is a particular commit you want to look back on and are not sure how far back it was, having to refer to it as `HEAD~n` could get extremely frustrating, especially since n will continue to change as you work more on your project and make more commits. Luckily there is another way to refer to a particular commit, using its commit number.
+
+When you enter `git commit -m "short descriptive message"` no output appears, but Git gives that commit a unique identifier, its commit number. By typing `git log` we can see ALL of the previous commits. Since that might be a lot of output to deal with, we will just ask for the last 3 commits using `git log -n 3`. The number after `-n` is how many commits you want displayed from the log.
+
+Let's see what that output looks like for Dracula:
+
+```
+$ git log -n 3
+commit 584977a18319f539f3f5ecde6176db2ba0081cd9 (HEAD -> main, origin/main)
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Mon Mar 14 14:59:40 2022 -0400
+
+    Ignore data files and the results folder.
+
+commit fe532b097861acb8bd3d7f221d6ee741249dc8f0
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Mon Mar 14 14:58:17 2022 -0400
+
+    Add concerns about effects of Mars' moons on Wolfman
+
+commit c3fe500d96fa9114210320f3cac0a0d9fd58d33c
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Mon Mar 14 14:57:09 2022 -0400
+
+    Start notes on Mars as a base
+```
+
+If you forget to include the `-n 3` flag and just type `git log` you can always jump to the end by pressing `q`. If you have been working on a project for a while, using the arrow keys to scroll down may take a prohibitively long time.
+
+The first thing to notice is that all of your commit messages are here. This is a good reminder to write clear and concise messages because future you may be very grateful when trying to figure out where exactly past you introduced a particular issue.
```

Comment:
```suggestion
The first thing to notice is that all of your commit messages are here. This is a good reminder to write clear and concise messages because **future** you may be very grateful when trying to figure out where exactly **past** you introduced a particular issue.
```